src/index.js

```diff
@@ -15,25 +17,75 @@ const traverse = require('traverse');
 const DEFAULT_REPLACEMENT = '--REDACTED--';
 
 /**
- * Module exports.
+ * Validate serializers.
  */
 
-module.exports = (
+function validateSerializers(serializers) {
+  serializers.map(({ path, serializer }) => {
+    if (typeof serializer !== 'function') {
+      throw new TypeError(`Invalid serializer for \`${path}\` path: must be a function`);
+    }
+  });
+}
+
+/**
+ * `parseAndSerialize` builds a copy of the original values and mutates the copy
+ *  with the serialized values.
+ */
+
+function parseAndSerialize(values, serializers) {
+  const target = JSON.parse(stringify(values));
+
+  for (const { path, serializer } of serializers) {
+    const value = get(values, path);
+
+    if (value === undefined) {
+      continue;
+    }
+
+    try {
+      set(target, path, serializer(value));
```

Comment:
Should we depend on serializers to not mutate the value? Or should we clone it before sending it in?